exec/java-exec/src/test/java/org/apache/drill/exec/physical/impl/join/TestMergeJoinAdvanced.java

```diff
@@ -255,6 +260,41 @@ public void testDrill4196() throws Exception {
       .go();
   }
 
+  @Test // DRILL-6491
+  public void testMergeIsNotSelectedForFullJoin() throws Exception {
+    String query = "select * " +
+        " from (select employee_id from cp.`employee.json` order by employee_id) e1 " +
+        " full outer join (select employee_id from cp.`employee.json` order by employee_id) e2 " +
+        " on e1.employee_id = e2.employee_id " +
+        " limit 10";
+    try {
+      setSessionOption(PlannerSettings.HASHJOIN.getOptionName(), true);
+      testPlanMatchingPatterns(query, new String[] {"HashJoin"}, new String[] {"MergeJoin"});
+    } finally {
+      resetSessionOption(PlannerSettings.HASHJOIN.getOptionName());
```

Comment:
This option is set in `@BeforeClass` method, but here you are resetting it to the default value. It may cause failures for other tests in the class which depends on the value of this option.